### PR TITLE
[DOCS] Clarify behavior of the generic `data` node role

### LINKED ISSUE #105000: node.roles: data takes precedence over specialised data roles
### Description

When we set node.role to [data](https://www.elastic.co/guide/en/elasticsearch/reference/8.12/modules-node.html#data-node) and specialised data roles, `data` role takes precedence over specialised data roles (`data_hot`, `data_warm`, `data_cold`, etc) affecting the behaviour of [index.routing.allocation.require._tier_preference](https://www.elastic.co/guide/en/elasticsearch/reference/current/data-tier-shard-filtering.html#data-tier-allocation-filters).

```
node.roles: ["data", "data_hot"]
node.roles: ["data", "data_warm"]
node.roles: ["data", "data_cold"]
```

Related Discuss Forum thread: https://discuss.elastic.co/t/can-both-allocate-this-shard-and-improve-the-cluster-balance/285579/5

### Doc Enhancement Request
[Documentation](https://www.elastic.co/guide/en/elasticsearch/reference/8.12/modules-node.html#node-roles) should explain the generic data roles takes precedence over specialised data role when being set together.

### PR DESCRIPTION
**Pull request by shainaraskas**
Fixes https://github.com/elastic/elasticsearch/issues/105000

Clarifies the behavior of the generic `data` node role, and added a warning explaining precedence rules for data roles

Also tweaked the IA a little to make all of the role descriptions children of a single "Data roles" heading, created a dedicated section for the generic `data` role type

**Preview**:
https://elasticsearch_bk_106375.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/modules-node.html

### CODE REVIEW

File: docs/reference/modules/node.asciidoc [hunk 1]
```diff
@@ -68,8 +68,8 @@ A node that has the `master` role, which makes it eligible to be
 
 <<data-node,Data node>>::
 
-A node that has the `data` role. Data nodes hold data and perform data
-related operations such as CRUD, search, and aggregations. A node with the `data` role can fill any of the specialised data node roles.
+A node that has one of several data roles. Data nodes hold data and perform data
+related operations such as CRUD, search, and aggregations. A node with a generic `data` role can fill any of the specialised data node roles.
```
Comment (shainaraskas) [inline docs/reference/modules/node.asciidoc:72]:
```suggestion
related operations such as CRUD, search, and aggregations. A node with a generic `data` role can fill any of the specialized data node roles.
```

**Comment by kingherc** [comment]
I believe @elastic/es-data-management is better suited to review this, so I'll change labels from the Distributed team to Data Management team. Feel free to undo if I'm mistaken.

File: docs/reference/modules/node.asciidoc (hunk 1 above)
Comment (joegallo) [reply docs/reference/modules/node.asciidoc:72]: +1 to preferring the American English spelling

**Review by joegallo — APPROVED** [review]
These changes make sense to me. 👍

**Review by karenzone — APPROVED** [review]

**Comment by shainaraskas** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.13|[](https://github.com/elastic/elasticsearch/pull/109690)|
|✅|8.12|[](https://github.com/elastic/elasticsearch/pull/109691)|
|✅|7.17|[](https://github.com/elastic/elasticsearch/pull/109692)|

Note: Successful backport PRs will be merged automatically after passing CI.

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sorenlouv/backport)

**Pull request merged**